instrumentation/kafka-streams-0.11/src/main/java/io/opentelemetry/auto/instrumentation/kafka_streams/KafkaStreamsProcessorInstrumentation.java

```diff
@@ -35,6 +32,10 @@
   // Also this instrumentation doesn't define 'producer' making it 'asymmetric' - resulting
   // in awkward tests and traces. We may want to revisit this in the future.
 
+  public static class SpanScopeHolder {
+    public static final ThreadLocal<SpanScopePair> CURRENT = new ThreadLocal<>();
```

Comment:
I don't think it can get out of sync, unless there are future changes to the kafka streams lib that invalidate the assumption of our instrumentation:

```
    public boolean process() {
        final StampedRecord record = partitionGroup.nextRecord(recordInfo);
        ...
        <on exit (success or failure), if CURRENT was set, then end span and clear>
    }

    StampedRecord nextRecord(RecordInfo info) {
        ...
        <on successful exit, start span and set CURRENT>
    }
```

It could be better though, I just pushed a change to make the instrumentation more future-proof (e.g. in case there are changes to kafka streams lib).